Issue 39696: Search reference for file breaks when renaming container

Sigmonia opened the pull request:
Fixing tests and other issues from https://github.com/LabKey/platform/pull/1041

Sigmonia committed:
Fix re-indexing issues
- Correct Since = null check
- Update list indexing to use since date to force reindex

Sigmonia committed:
include missed file

Sigmonia committed:
Attempt to clear only the documents related to a container's files
- Added new deleting method to LuceneSearchServiceImpl

labkey-tchad commented on core/src/org/labkey/core/CoreContainerListener.java:
```diff
@@ -106,14 +106,14 @@ public void propertyChange(PropertyChangeEvent propertyChangeEvent)
-                String message = evt.container.getName() + " was renamed from " + oldValue + " to " + newValue;
-                addAuditEvent(evt.user, evt.container, message);
+                String message = c.getName() + " was renamed from " + oldValue + " to " + newValue;
+                addAuditEvent(evt.user, c, message);
```
Why doesn't the reindexing get triggered in `SearchContainerListener`? That's where we trigger the initial crawl for new conainers (`DavCrawler.getInstance().addPathToCrawl`)

labkey-tchad reviewed (changes requested):
This isn't working for me locally. It also appears to only address folder rename but not move.

labkey-tchad committed:
More targeted reindexing

Moved the triggers to SearchContainerListener
Work for container move in addition to rename

labkey-tchad commented:
They still aren't getting reindexed but at least we're only clearing the index for actual files now (12 search failures instead of 15).

labkey-tchad reviewed (approved):
This is an improvement and should resolve failures in Sample Manager tests. We can continue work in a new branch.
